components/clarity-repl/src/repl/datastore.rs

```diff
@@ -319,193 +325,316 @@ impl ClarityBackingStore for Datastore {
+            consensus_hash,
+            vrf_seed,
+            stacks_block_time,
+        }
+    }
+
+    pub fn advance_burn_chain_tip(
+        &mut self,
+        clarity_datastore: &mut ClarityDatastore,
+        count: u32,
+    ) -> u32 {
+        for _ in 1..=count {
+            let last_stacks_block = self
+                .stacks_blocks
+                .get(&clarity_datastore.current_chain_tip)
+                .unwrap();
+            let last_burn_block = self
+                .burn_blocks
+                .get(&last_stacks_block.burn_block_header_hash)
+                .unwrap();
+
+            let mut next_burn_block_time = last_burn_block.burn_block_time + 600;
+            if last_stacks_block.stacks_block_time > next_burn_block_time {
+                next_burn_block_time = last_stacks_block.stacks_block_time + 10;
+            }
+
+            let height = self.burn_chain_height + 1;
+            let hash = height_to_burn_block_header_hash(height);
+            let burn_block_info = BurnBlockInfo {
+                burn_block_time: next_burn_block_time,
+                burn_block_height: height,
+            };
+
+            self.burn_blocks.insert(hash, burn_block_info);
+            self.burn_chain_height = height;
+            self.advance_stacks_chain_tip(clarity_datastore, 1);
+        }
+
+        self.burn_chain_height
+    }
+
+    pub fn advance_stacks_chain_tip(
+        &mut self,
+        clarity_datastore: &mut ClarityDatastore,
+        count: u32,
+    ) -> u32 {
+        let current_lookup_id = *clarity_datastore
             .block_id_lookup
-            .get(&self.open_chain_tip)
+            .get(&clarity_datastore.open_chain_tip)
             .expect("Open chain tip missing in block id lookup table");
-        let genesis_time = self.genesis_time;
 
-        for i in 1..=count {
-            let height = cur_height + i;
-            let bytes = height_to_hashed_bytes(height);
+        for _ in 1..=count {
+            self.stacks_chain_height += 1;
+
+            let bytes = height_to_hashed_bytes(self.stacks_chain_height);
             let id = StacksBlockId(bytes);
             let sortition_id = SortitionId(bytes);
-            let block_info = height_to_block(height, Some(genesis_time));
-            self.block_id_lookup.insert(id, current_lookup_id);
-            self.height_at_chain_tip.insert(id, height);
+            let block_info = self.build_next_stacks_block(clarity_datastore);
+
             self.sortition_lookup.insert(sortition_id, id);
             self.consensus_hash_lookup
                 .insert(block_info.consensus_hash, sortition_id);
-            self.store.insert(id, block_info);
+            self.stacks_blocks.insert(id, block_info);
+
+            clarity_datastore
+                .block_id_lookup
+                .insert(id, current_lookup_id);
+            clarity_datastore
+                .height_at_chain_tip
+                .insert(id, self.stacks_chain_height);
```

Comment:
These operations, inserting into hashmaps and updating states, assume no failure. Error handling could be improved to handle cases where these operations fail.

Reply:
Addressed